internal/resources/oncall/resource_escalation.go

```diff
@@ -216,6 +217,21 @@ func resourceEscalation() *common.Resource {
 				},
 				Description: "The end of the time interval for notify_if_time_from_to type step in UTC (for example 18:00:00Z).",
 			},
+			"severity": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ConflictsWith: []string{
+					"duration",
+					"notify_on_call_from_schedule",
+					"persons_to_notify",
+					"persons_to_notify_next_each_time",
+					"notify_to_team_members",
+					"notify_if_time_from",
+					"notify_if_time_to",
+					"action_to_trigger",
+				},
+				Description: "The severity of the incident for declare_incident type step.",
+			},
```

Comment:
the value of this is essentially an enum, right? If so, I'd suggest adding a `ValidateFunc` here, like [this](https://github.com/grafana/terraform-provider-grafana/pull/1836/files#diff-f690fc0017aaf007bb7b6f788fc54da577d98cc99a7960d9def49d579a8f2ae9R67-R72):
```go
ValidateFunc: validation.StringInSlice(severityOptions, false),
Description:  fmt.Sprintf("The severity of the incident for declare_incident type step. Can be %s", severityOptionsVerbal),
```

Reply:
We have similar enumerations for each possible param, but each list the specific conflicting params which are different in each case. I think for now I would prefer having the list explicit and readable for each param? (I can take a look at defining multiple enums and refactoring in a separate PR too)

Reply:
ohh I see, sounds more involved, was just curious 👍